<head>
  <title>ViroidDB</title>
  <link rel="stylesheet" href="dist/tailwind.css" />
</head>

<body>
  <div class="prose mx-auto my-10 md:my-20">
    <h1>ViroidDB</h1>
    <p>
      Viroids and their (potential) relatives are among the most unique and
      exciting of all biological entities. However, I couldn't find a good,
      up-to-date database containing all of them in one place so I made my own.
    </p>

    <p>
      This database has sequences representing known viroid and viroid-like RNA
      sequences. Each sequence is unique, though variants of the same RNA
      species are included. Currently, sequences are <i>not</i> guaranteed to be
      rotationally unique, meaning that <code>AAAATTTT</code> is treated as
      distinct from <code>TTAAAATT</code>.
    </p>
    <h2>How to get the data</h2>
    <p>
      You can download the most recent version of the entire database as a FASTA
      file <a>here</a>.
    </p>
    <p>
      Alternatively, download the database easily using your command-line tool
      of choice:
    </p>
    <pre>$ curl viroids.org/db/all.fasta > viroiddb.fasta</pre>

    <p>
      If you're interested in a
      <abbr title="noting, of course, that a set is a subset of itself">
        subset
      </abbr>
      of the data, here are all of available subsets:
    </p>
    <table>
      <thead>
        <td>Subset</td>
        <td>Download link</td>
        <td>Sequence count</td>
        <td>Total bases</td>
      </thead>
      <tbody>
        <tr>
          <td>All</td>
          <td><a href="">/db/all.fasta</a></td>

          <td>9,891</td>
          <td>4,001,024</td>
        </tr>
        <tr>
          <td>Viroids</td>
          <td><a href="">/db/viroids.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
        <tr>
          <td style="text-indent: 1em"><i>Avsunviroidae</i></td>
          <td><a href="">/db/avsunviroidae.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
        <tr>
          <td style="text-indent: 1em"><i>Pospiviroidae</i></td>
          <td><a href="">/db/pospiviroidae.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
        <tr>
          <td style="text-indent: 1em">Unclassified</td>
          <td><a href="">/db/unclassified.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
        <tr>
          <td><i>Deltavirus</i></td>
          <td><a href="">/db/deltavirus.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
        <tr>
          <td>Retrozymes</td>
          <td><a href="">/db/retrozymes.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
        <tr>
          <td>Satellite RNAs</td>
          <td><a href="">/db/satellites.fasta</a></td>
          <td></td>
          <td></td>
        </tr>
      </tbody>
    </table>

    <p>
      Archived versions of data are available for each release. Releases do not
      currently happen at a set interval due to the slow pace of viroid
      discovery, though I do hope to change that in the future. Previous
      releases can be accessed through URLs of the form
      <code>https://viroids.org/db/YYYY-MM-DD/subset.fasta</code>. Available
      releases:
    </p>
    <ul>
      <li>2021-04-15</li>
    </ul>

    <h2>How I curate the database</h2>
    <p>
      All data is publicly available from the NCBI. I have included what I
      believe to be a class of potentially related RNA agents that merit study
      as a group. The key inclusion criteria are:
    </p>
    <ul>
      <li>Circularity at some point in the replication cycle</li>
      <li>Extensive RNA folding</li>
      <li>The presence of ribozymes or a central conserved region</li>
    </ul>
    <p>To collect the sequences for the database I followed this protocol:</p>
    <ol>
      <li>
        For viroids and deltaviruses, I used the NCBI Virus portal to download
        complete sequences within the taxonomy IDs 185751, 185752, 265963 and
        39759 for the <i>Pospiviroidae</i>, <i>Avsunviroidae</i>, unclassified
        viroids, and <i>Deltavirus</i> taxa respectively.
      </li>
      <li>
        For retrozymes, which are not taxonomically classified, I downloaded all
        sequences within the
        <a href="https://www.ncbi.nlm.nih.gov/nucleotide/">NCBI Nucleotide</a>
        database matching the search term "ribozyme".
      </li>
      <li>
        Similarly, for satellite RNAs, I downloaded all sequences within the
        <a href="https://www.ncbi.nlm.nih.gov/nucleotide/">NCBI Nucleotide</a>
        database matching each of the small circular satellites in
        <a
          href="https://www.elsevier.com/books/viroids-and-satellites/hadidi/978-0-12-801498-1"
          ><i>Viroids and Satellites</i> (2017)</a
        >.
      </li>
      <li>
        Deduplication and formatting is performed by
        <a href="https://bioinf.shenwei.me/seqkit/">SeqKit</a>.
      </li>
    </ol>
    <p>
      All the processing (and website updating) is currently done manually.
      Until NCBI Virus has an API, I will repeat this process occasionally and
      update this site. Should new viroid-like RNAs classes be discovered, I
      will add them as a new subset. If you feel an update is required, please
      <a href="#contact">contact me</a> or perform the update yourself and
      submit it as a pull request to the repository.
    </p>
    <h2 id="contact">How to contact me</h2>
    <p>
      Feel free to contact me via sending an email or filing an issue on GitHub.
    </p>
    <hr />
    <p>
      Brought to you by
      <a href="https://benjamindlee.com" target="_blank">Benjamin Lee</a> at
      NCBI/NLM/NIH & University of Oxford.
    </p>
  </div>
</body>
